pkg/types/ticket/custom_fields.go

```diff
@@ -0,0 +1,586 @@
+package ticket
```

Comment:
So much of this could be generated. Maybe worth writing a generator if this is the direction we decide to go.